node/libs/crypto/src/bls12_381/tests.rs

```diff
@@ -16,6 +16,18 @@ fn signature_smoke() {
     sig.verify(&msg, &pk).unwrap()
 }
 
+#[test]
+fn infinity_public_key_failure() {
+    let err = PublicKey::decode(&INFINITY_PUBLIC_KEY)
+        .expect_err("Decoding the infinity public key should fail");
+
+    assert_eq!(
+        err.to_string(),
+        Error::InvalidInfinityPublicKey.to_string(),
+        "Unexpected error message for infinity public key decoding"
+    );
```

Comment:
don't check human readable messages for equality. use assert_match instead.

Reply:
or just check that parsing fails without checking the error variant, given that the function returns anyhow::Error.

Reply:
ok